components/pkg-export-kubernetes/src/manifest.rs

```diff
@@ -151,13 +154,20 @@ impl Manifest {
                       environment })
     }
 
-    /// Generates the manifest as a string and writes it to `write`.
+    /// Generates manifest as a string and writes it to `write`.
     pub fn generate(&mut self, write: &mut dyn Write) -> Result<()> {
         let out: String = ManifestJson::new(&self).into();
 
         write.write_all(out.as_bytes())?;
         Ok(())
     }
+
+    // Replaces any periods in the metadata name with hyphens
+    // To make it a valid Kubernetes resource name
+    fn formatted_metadata_name(metadata_name: &str) -> String {
+        let re = Regex::new(r"\.").unwrap();
+        re.replace_all(metadata_name, "-").to_string()
+    }
```

Comment:
Using a `Regex` to replace periods with hyphens feels a little heavyweight, in my opinion.  You could accomplish the same thing with `name.replace(".", "-")`.

Reply:
That's an excellent point, making that change now.

Reply:
Done!